File: PB - More Exercises GO/1.0 Simple Operations and Calculations - More Exercises/2.0 Conditional Statements - More Exercises/01. Pipes In Pool.go
package main

import "fmt"

func main() {
	var volume, pipeOne, pipeTwo, hours float64
	fmt.Scan(&volume, &pipeOne, &pipeTwo, &hours)
	pipeOne *= hours
	pipeTwo *= hours
	pool := pipeOne + pipeTwo
	poolPercent := (pool / volume) * 100
	pipeOnePercent := (pipeOne / pool) * 100
	pipeTwoPercent := (pipeTwo / pool) * 100

	if pool <= volume {
		fmt.Printf("The pool is %.2f%% full. Pipe 1: %.2f%%. Pipe 2: %.2f%%.", poolPercent, pipeOnePercent, pipeTwoPercent)
	} else {
		fmt.Printf("For %.2f hours the pool overflows with %.2f liters.", hours, pool-volume)
	}

}

/*
 name   :01. Pipes In Pool
 input  :
1000
100
120
3
output : The pool is 66.00% full. Pipe 1: 45.45%. Pipe 2: 54.55%.
*/

File: PB - More Exercises Python/2.0 Conditional Statements - More Exercises/01. Pipes In Pool.py
volume, pipe_one, pipe_two, hours = [float(input()) for _ in range(4)]
pipe_one *= hours
pipe_two *= hours
pool = pipe_one + pipe_two
pool_percent = pool / volume * 100
pipe_one_percent = pipe_one / pool * 100
pipe_two_percent = pipe_two / pool * 100

if pool <= volume:
    print(f'The pool is {pool_percent:.2f}% full. Pipe 1: {pipe_one_percent:.2f}%. Pipe 2: {pipe_two_percent:.2f}%.')
else:
    print(f'For ${hours:.2f} hours the pool overflows with {pool - volume:.2f} liters.')



'''
1000
100
120
3
'''
